scripts/deploy_infra.py

```diff
@@ -299,11 +299,11 @@ def deploy_infra(network, url, account, fork=False):
         if not fork:
             boa_zksync.set_zksync_env(url)
             logger.log("Prodmode on zksync Era ...")
-            boa.env.add_account(Account.from_key(os.environ[account]))
         else:
             boa_zksync.set_zksync_fork(url)
             logger.log("Forkmode on zksync Era ...")
-            boa.env.eoa = deploy_utils.FIDDYDEPLOYER  # set eoa address here
+
+        boa.env.set_eoa(Account.from_key(os.environ[account]))
```

Comment:
I believe set_eoa expects only the address, not the account
The current way to do this is `boa.env.add_account(Account.., force_eoa=True)`